lib/SIL/Verifier/MemoryLifetime.cpp

```diff
@@ -153,7 +153,8 @@ void MemoryLocations::analyzeLocations(SILFunction *function) {
   for (SILBasicBlock &BB : *function) {
     for (SILInstruction &I : BB) {
       auto *ASI = dyn_cast<AllocStackInst>(&I);
-      if (ASI && !ASI->hasDynamicLifetime()) {
+      if (ASI && !ASI->hasDynamicLifetime() &&
+          !ASI->getType().getEnumOrBoundGenericEnum()) {
```

Comment:
This line needs a comment. You can copy the commit message here. Please also file a bug report for the to-do (you can assign it to me) and reference it in the comment.

Reply:
Will do! I was going to talk with you before this went in!

Reply:
(Wanted to get everything to pass first).

Reply:
@eeckstein I filed https://bugs.swift.org/browse/SR-14123 and put that into a comment with the commit msg. I also assigned the SR to you.